tensorrt_llm/_torch/visual_gen/executor.py

```
@@ -318,17 +319,58 @@ def process_request(self, req: DiffusionRequest):
+        """Return whether per-request CUDA peak memory logging is enabled."""
+        pipeline_config = getattr(self.diffusion_args, "pipeline", None)
+        return bool(getattr(pipeline_config, "enable_cuda_memory_logging", False))
+
+    def _reset_cuda_peak_memory_stats(self) -> None:
+        """Reset CUDA peak memory stats for this worker if CUDA is available."""
+        if not torch.cuda.is_available():
+            return
+
+        try:
+            torch.cuda.reset_peak_memory_stats(self.device_id)
+        except Exception as e:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)